### PR TITLE
Create Selection Field Component

### LINKED ISSUE #260: Create SelectionField component
## Description
For some settings (like groupTypes or deckTypes) we need a SelectionField component

## Steps
- [ ] Create new component in `/components/input/SelectionField.tsx` that
  - [ ]  should accept a list of options (icon, title, description) as input

## Design
Have a look at Figma for the design.

Screenshot 2024-02-15 at 13 46 26

## Next steps
This component should be used in the Group- and DeckModal as described in these issues
- #250
- #251

### PR DESCRIPTION
**Pull request by leonie2003**
## Description
Add Selection Field Component that accepts a list of options (icon, title, description) as input.

## Resolved Issues
Resolves #260

### CODE REVIEW

File: frontend/components/input/SelectionField.tsx [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+									setListSelected(list, selectionItem);
+								}}
+							>
+								<Icon
+									className="w-10 pr-3"
+									icon={selectionItem.icon}
+								></Icon>
+								<div>
+									<Text className="font-bold">
+										{" "}
```
Comment (Julian702) [inline frontend/components/input/SelectionField.tsx:88]: Is this necessary?

**Review by Julian702 — CHANGES_REQUESTED** [review]
Looks great already, here are my first comments on some details :)

File: frontend/components/input/SelectionField.tsx [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+	);
+	return (
+		<div>
+			<Text className="mb-1 text-sm font-semibold text-neutral-400">
+				{title}
+			</Text>
+			<div
+				className="flex	hover:cursor-pointer"
+				role="button"
+				onKeyDown={() => {}}
```
Comment (Julian702) [inline frontend/components/input/SelectionField.tsx:50]: I think this is a change resulting from a sonarcloud issue. Im not sure about the requirement here, but I think `combobox` would be the proper choice for the role. So the result should be something like this:
```suggestion
				role="combobox"
				onKeyUp={clickOnEnter}
				tabIndex={0}
```
Note that you can use the util function `clickOnEnter` to dispatch a click event on pressing enter.

PS: You can do something similar for the option with role `option`.

File: frontend/components/input/SelectionField.tsx [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+	const [selected, setSelected] = useState<SelectionListItem>(
+		getListSelected(list)
+	);
+	return (
+		<div>
+			<Text className="mb-1 text-sm font-semibold text-neutral-400">
+				{title}
+			</Text>
+			<div
+				className="flex	hover:cursor-pointer"
```
Comment (Julian702) [inline frontend/components/input/SelectionField.tsx:48]: Hover state is not necessary here
```suggestion
				className="flex cursor-pointer"
```

File: frontend/components/input/SelectionField.tsx [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+					{list?.map((selectionItem) => (
+						<div
+							key={selectionItem.title}
+							className={`flex hover:cursor-pointer ${
+								!selectionItem.isSelected
+									? "text-neutral-400 hover:text-neutral-300"
+									: ""
+							}`}
+							role="button"
+							onKeyDown={() => {}}
```
Comment (Julian702) [inline frontend/components/input/SelectionField.tsx:76]: Use `option` here

**Review by Julian702 — CHANGES_REQUESTED** [review]
I have some more comments on this. Moreover I think we should avoid fixed widths in the component and define the width by setting additional classnames in the parent.

**Comment by Julian702** [comment]
We also need a story for the new component.

**Review by Julian702 — APPROVED** [review]

**Review by TomRomeo — APPROVED** [review]

**Pull request merged**